packages/dd-trace/src/encode/0.4.js

```diff
@@ -263,6 +271,45 @@ class AgentEncoder {
     }
   }
 
+  _encodeObject (bytes, value, circularReferencesDetector = new Set()) {
+    circularReferencesDetector.add(value)
+    if (Array.isArray(value)) {
+      return this._encodeObjectAsArray(bytes, value, circularReferencesDetector)
+    } else if (typeof value === 'object') {
+      return this._encodeObjectAsMap(bytes, value, circularReferencesDetector)
```

Comment:
```javascript
typeof null === 'object' // true
// but
Object.keys(null) // TypeError: Cannot convert undefined or null to object

Reply:
Good catch, I will add a test and fix this case.